chopper_generator/lib/src/generator.dart

```diff
@@ -466,6 +485,43 @@ final class ChopperGenerator
     });
   }
 
+  static Reference? _generateCleanMapBody(
+    List<Code> blocks,
+    String refName, {
+    bool valueToString = false,
+    DartType? type,
+  }) {
+    final bool handleNullableValue;
+    final bool handleToString;
+    if (type != null && !_isMapStringDynamic(type)) {
+      handleNullableValue = false;
+      handleToString = false;
+    } else {
+      handleNullableValue = true;
+      handleToString = valueToString;
+    }
+
+    if (!handleToString && !handleNullableValue) {
+      return null;
+    }
+
+    final bodyType = type?.toString() ?? 'Map<String, dynamic>';
+    final cleanType = valueToString ? 'Map<String, String>' : bodyType;
+    final cleanSymbol = '${refName}Clean';
+    blocks.add(Code('final $cleanType $cleanSymbol = {};'));
+    blocks.add(Code('for (var key in $refName.keys) {'));
+    if (handleNullableValue) {
```

Comment:
Not sure if nullable need to be checked. When looking at the "old" `FormUrlEncodedConverter` it doesn't care about null values.

Reply:
> Not sure if nullable need to be checked. When looking at the "old" `FormUrlEncodedConverter` it doesn't care about null values.

null values are meaningless in most scenarios, maybe add an optional bool switch？